### PR TITLE
Rc2 fixes

### LINKED ISSUE #302: Update build.cmd to use latest nuget
build.cmd downloads the nuget.exe from https://nuget.org/nuget.exe. That does not download the latest version. Use https://dist.nuget.org/win-x86-commandline/latest/nuget.exe instead.

Also, update nuget.config to use v3 feeds (both myget and nuget), they are faster than v2 feeds. KRelease branch has those changes:
    <add key="DotNetRelease" value="https://www.myget.org/F/dotnet-core-rel/api/v3/index.json" />
    <add key="AspNetRelease" value="https://www.myget.org/F/aspnetcirelease/api/v3/index.json" />
    <add key="NuGet.org" value="https://api.nuget.org/v3/index.json" />

### PR DESCRIPTION
**Pull request by tushargupta51**
Fixes #302, #289

### CODE REVIEW

**Comment by tushargupta51** [comment]
@brentschmaltz

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs [hunk 1]
```diff
@@ -40,13 +41,26 @@ public class OpenIdConnectMessage : AuthenticationProtocolMessage
...
 
         /// <summary>
-        /// Initializes an instance of <see cref="OpenIdConnectMessage"/> class with a specific issuerAddress.
+        /// Initializes an instance of <see cref="OpenIdConnectMessage"/> class with a json string.
         /// </summary>
-        public OpenIdConnectMessage(string issuerAddress) : base(issuerAddress) 
+        public OpenIdConnectMessage(string json)
         {
+            if (string.IsNullOrEmpty(json))
+                LogHelper.LogArgumentNullException("json");
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs]: need to do "throw ..." as the api just returns a populated exception.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs [hunk 2]
```diff
@@ -40,13 +41,26 @@ public class OpenIdConnectMessage : AuthenticationProtocolMessage
...
+
+            try
+            {
+                JObject jsonObject = JObject.Parse(json);
+                SetJsonParameters(jsonObject);
+            }
+            catch
+            {
+                LogHelper.LogException<ArgumentException>(LogMessages.IDX10106, json);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs]: same, need to 'throw...'

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs (hunk 1 above)
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs]: right, forgot that. will update

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs [hunk 3]
```diff
@@ -40,13 +41,26 @@ public class OpenIdConnectMessage : AuthenticationProtocolMessage
...
-        public OpenIdConnectMessage(string issuerAddress) : base(issuerAddress) 
+        public OpenIdConnectMessage(string json)
         {
+            if (string.IsNullOrEmpty(json))
+                LogHelper.LogArgumentNullException("json");
+
+            try
+            {
+                JObject jsonObject = JObject.Parse(json);
+                SetJsonParameters(jsonObject);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs]: You could choose to write: SetJsonParameters(JObject.Parse(json)); 
Loosing the local variable.

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs [hunk 4]
```diff
@@ -45,12 +45,13 @@ public void Constructors()
         {
             OpenIdConnectMessage openIdConnectMessage = new OpenIdConnectMessage();
             Assert.Equal(openIdConnectMessage.IssuerAddress, string.Empty);
-            openIdConnectMessage = new OpenIdConnectMessage("http://www.got.jwt.com");
+            openIdConnectMessage = new OpenIdConnectMessage() { IssuerAddress = "http://www.got.jwt.com" };
             Assert.Equal(openIdConnectMessage.IssuerAddress, "http://www.got.jwt.com");
-            ExpectedException expectedException = ExpectedException.ArgumentNullException("IssuerAddress");
+            ExpectedException expectedException = ExpectedException.NoExceptionExpected;
+            string json = @"{""response_mode"":""responseMode"", ""response_type"":""responseType"", ""refresh_token"":""refreshToken""}";
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMessageTests.cs]: We should have a tests for 'bad' json. Maybe there is, but I don't see it here.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/LogMessages.cs [hunk 5]
```diff
@@ -40,6 +40,7 @@ internal static class LogMessages
 
         // properties, configuration 
         internal const string IDX10105 = "IDX10105: NonceLifetime must be greater than zero. value: '{0}'";
+        internal const string IDX10106 = "IDX10106: Error in deserializing json: '{0}' into Newtonsoft.JObject.";
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/LogMessages.cs]: We shouldn't specify 'Newtonsoft' here. Just inform user that the 'json' was not well formed.

Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/LogMessages.cs]: We are specifically trying to convert the json string to Newtonsoft.JObject. What is the harm in specifying 'Newtonsoft'?

**Pull request merged**